docs/debugging-testing/testing/unit_testing.md

```diff
@@ -227,40 +202,55 @@ With the following steps, you can write a unit test. This example creates dummy
     }
     ```
 
-1. Stub all external dependencies. Create the following stub in the Mbed OS root directory `UNITTESTS/stubs`:
-
-    **OtherClass_stub.cpp**
-
+    ### CMake input source file for UUT
+    ```
+    set(TEST_NAME myclass-unittest)
+
+    add_executable(${TEST_NAME})
+  
+    target_sources(${TEST_NAME}
+        PRIVATE
+            ${MBED_PATH}/example/tests/unittests/MyClass/MyClass.cpp
+            test_MyClass.cpp
+    )
+    
+    target_link_libraries(${TEST_NAME}
+        PRIVATE
+            mbed-headers
+            mbed-stubs            
+            gmock_main
+    )
+    
+    add_test(NAME "${TEST_NAME}" COMMAND ${TEST_NAME})
+    
+    set_tests_properties(${TEST_NAME} PROPERTIES LABELS "example")
     ```
-    #include "example/OtherClass.h"
 
-    namespace example {
+This example does not use any Mbed OS code. If your unit tests do, remember to update header stubs in `UNITTESTS/target_h` and stubs sources in `UNITTESTS/stubs` with any missing type or function declarations.
 
-    int OtherClass::otherFunction() {
-        return 0;
-    }
+### Building and running unit tests
 
-    }
-    ```
+#### Mbed CLI 1
 
-This example does not use any Mbed OS code, but if your unit tests do, then remember to update header stubs in `UNITTESTS/target_h` and source stubs in `UNITTESTS/stubs` with any missing type or function declarations.
+- Install [Mbed CLI 1](../build-tools/mbed-cli-1.html) with `pip install mbed-cli` command in Debian or Ubuntu or macOS or Windows and make sure that installed Mbed CLI 1 version 1.8.0 or newer.
 
-### Building and running unit tests
+- Mbed CLI 1 supports unit tests through the `mbed test --unittests` command. To learn how to use unit tests with Mbed CLI 1, please see the [unit testing documentation section](../build-tools/test-and-debug.html). For other information on using Mbed CLI 1, please see the [CLI documentation in handbook](../build-tools/mbed-cli-1.html).
+
+#### Mbed CLI 2
 
-Use Mbed CLI to build and run unit tests. For advanced use, you can run CMake and a Make program directly.
+- Mbed CLI 2 does not currently support the `mbed test --unittests` command, please use CMake and a Make command directly.
 
 #### Build tests directly with CMake
 
 1. Create a build directory: `mkdir UNITTESTS/build`.
 1. Move to the build directory: `cd UNITTESTS/build`.
-1. Run CMake using a relative path to the `UNITTESTS` folder as the argument. So from `UNITTESTS/build` use `cmake ..`:
+1. Run CMake using a relative path to the `UNITTESTS` folder as the argument. So from `UNITTESTS/build` use `cmake ..-DMBED_BUILD_UNITTESTS=ON`:
```

Comment:
Is the cmake command correct? I see the check for MBED_BUILD_UNITTESTS in mbed-os/CMakeLists, which would imply we need to run a cmake configure from the mbed-os root and not the UNITTEST directory. When I follow the steps as written no unit tests are discovered.

Reply:
Cmake command is missing with one more change directory (`cd UNITTESTS/build`) in the  `cmake ../.. -DMBED_BUILD_UNITTESTS=ON` I will do the correction